src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -377,11 +377,19 @@ def _update_artifacts(artifacts, lab_resource_id):
 
     result_artifacts = []
     for artifact in artifacts:
-        artifact_id = artifact.get('artifact_id', None)
+        if artifact.__class__.__name__ == "ArtifactInstallProperties":
+            if len(artifact.parameters) > 0:
+                raise CLIError("One or more of the formula artifacts contains parameters. Use --artifacts to supply parameter values")
```

Comment:
Suggest:
Formulas with parameterized artifacts are not supported. Use --artifacts to supply artifacts to the virtual machine.

Reply:
fixed